### PR TITLE
Auto-load rotated log history

### PR DESCRIPTION
**Pull request by jan-moravec**
Opening a primary log now automatically discovers and loads its rotated history. Numbered logrotate families (`app.log.1`, `app.log.2.gz`, …) and dated variants (`app.log-2025-04-28`, `app-2025-04-28.log.zst`, …) are recognized, sorted oldest-first, and merged with the selected primary. Static opens, drag-and-drop, command-line opens, and Stream Mode share the same expansion path; Stream Mode drains the historical prefix before attaching the live tail so rows remain chronologically ordered across the handoff.

The feature is advisory and reversible. **Settings → Auto-detect rotated log history** persists the default, `--no-rotation-history` disables discovery for a launch (including launches forwarded through the single-instance protocol), and each session records `Source::followRotationSiblings`. When files were added automatically, the status bar reports the expansion and **Settings → Undo rotated history expansion** reopens the caller's original selection while preserving whether the request was static or live-tail.

**Rotation-family detection lives in `loglib`.** `rotation_siblings.hpp` recognizes numbered and dated suffixes with optional `.gz`, `.bz2`, `.xz`, and `.zst` codec extensions, derives the primary from any family member, computes stable canonical deduplication keys, and partitions mixed selections without disturbing unrelated caller order. Directory enumeration is capped at 4096 entries, compressed primaries are not expanded, pathological numeric suffixes are rejected, and all filesystem path conversion remains UTF-8-safe on Windows. Canonical keys intentionally match the app-layer locator contract so appends, symlinked paths, case rules, and restored sessions do not accumulate duplicate rows.

**The static-prefix to live-tail seam is explicit.** `LogModel::AppendStreaming(StreamLineSource)` lets `MainWindow` stream each historical file through the normal parser pipeline and then attach the primary producer without resetting retained rows. Promotion is queued outside teardown stacks, all-failed sibling prefixes rescue the original tail request, destructive opens scrub pending producers and undo state, and exceptions during tail attachment demote the UI cleanly instead of leaving a phantom live session. Retention remains bounded: when the live tail is armed, the configured cap FIFO-evicts excess historical rows.

**Single-instance forwarding now preserves launch intent.** The wire protocol advances to v3 with a `LaunchFlags` bitmask carrying `--no-rotation-history`; v2 frames remain accepted. A consumed-frame guard prevents buffered post-frame bytes from decoding the same open request twice.

Ships roadmap item **11**, [Pulling rotated history off disk](ROADMAP.md#11-pulling-rotated-history-off-disk), and documents discovery, ordering, opt-out, undo, persistence, and Stream Mode behavior in the user guide.

## Why

Rotated files are one logical stream, but opening `app.log` previously showed only the newest fragment. Investigations therefore began with manually finding siblings, selecting them in the right order, and repeating that work every time the active file rolled. The gap was especially visible in Stream Mode: tailing the primary omitted the history needed to explain the first live event.

The implementation keeps filesystem naming policy in the Qt-free library while leaving session orchestration in `MainWindow`. It does not watch arbitrary sibling paths or guess across directories; it recognizes bounded, conventional rotation families adjacent to the selected primary, then feeds those paths through the existing parsing, decompression, retention, and session machinery.

## Test surface

- **`test/lib/src/test_rotation_siblings.cpp`** covers numbered and dated families, codec suffixes, primary derivation, chronological sorting, mixed selections, caller-listed origins, duplicate suppression, platform case behavior, numeric edge cases, compressed primaries, non-ASCII paths, canonical keys, symlinks, and bounded directory enumeration.
- **`test/app/src/main_window_test.cpp`** covers static and Stream Mode expansion, prefix-to-tail promotion, all-sibling-failure rescue, teardown reentrancy, retention, undo, multi-file drops, append/replace deduplication, session persistence, menu/CLI overrides, non-ASCII paths, stale-state cleanup, toast anchoring, and configuration-UI gating.
- **Single-instance tests** cover v3 launch-flag forwarding, v2 compatibility, partial frames, and one-shot frame consumption.

## Test plan

- [x] `cmake --workflow --preset release` green locally on Windows + MSVC (699 tests).
- [x] `run-clang-tidy` over the affected translation units reports no project-code warnings after the clang-tidy sweep.
- [x] `pre-commit run --all-files` clean locally.
- [x] CI green on Linux, macOS, Windows, sanitizers, and coverage.

### CODE REVIEW

File: library/src/rotation_siblings.cpp [hunk 1]
```diff
@@ -0,0 +1,575 @@
...
+    // A codec extension cannot serve as the primary extension; treating it
+    // as one would misclassify companions of the uncompressed file.
+    if (IsRecognisedCodecExt(ext))
+    {
+        return candidates;
+    }
+    const SiblingMatchers matchers = BuildMatchers(primaryBasename, stem, ext);
+
+    // Use the same platform case rules for matching and primary exclusion.
+    const std::string primaryBasenameKey =
```
[the remark this answers is not in the document]
Comment (jan-moravec) [reply library/src/rotation_siblings.cpp]: Fixed in 4d9567e by binding the platform-specific key expression to a const reference, avoiding the Linux copy while preserving temporary lifetime on case-folding platforms.

File: library/src/rotation_siblings.cpp [hunk 2]
```diff
@@ -0,0 +1,575 @@
...
+    while (it != end)
+    {
+        if (entriesWalked++ >= MAX_ENUMERATED_DIRECTORY_ENTRIES)
+        {
+            break;
+        }
+        const std::filesystem::directory_entry &entry = *it;
+        const std::filesystem::path path = entry.path();
+        const std::string filename = PathToUtf8(path.filename());
+        const std::string filenameKey =
```
[the remark this answers is not in the document]
Comment (jan-moravec) [reply library/src/rotation_siblings.cpp]: Fixed in 4d9567e by binding the filename key expression to a const reference; the scoped rotation-sibling tests remain green.

File: library/src/rotation_siblings.cpp [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+
+/// Bounds the synchronous directory scan. Results may be truncated.
+constexpr std::size_t MAX_ENUMERATED_DIRECTORY_ENTRIES = 4096;
+
+/// Match case-insensitively on Windows/macOS and sensitively elsewhere.
+constexpr auto REGEX_CASE_FLAGS =
+#if defined(_WIN32) || defined(__APPLE__)
+    std::regex::icase;
+#else
+    std::regex::flag_type{};
```
[the remark this answers is not in the document]
Comment (jan-moravec) [reply library/src/rotation_siblings.cpp]: Fixed in 4d9567e by making ECMAScript the baseline regex syntax flag and OR-ing icase on Windows/macOS, eliminating the invalid zero enum value.
